storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -186,6 +191,12 @@ public Optional<String> serverConfigName(String configName) {
     @SuppressWarnings("deprecation")
     private static final String MESSAGE_FORMAT_VERSION_CONFIG = TopicConfig.MESSAGE_FORMAT_VERSION_CONFIG;
 
+    @SuppressWarnings("deprecation")
```

Comment:
what "deprecation" warning are we suppressing here? Wouldn't these configuration continue to exist in future?

Reply:
This is for `TopicConfig.MESSAGE_TIMESTAMP_DIFFERENCE_MAX_MS_CONFIG` It will be removed in the 4.0. I have added a comment to make it clear.